src/overlays/gamestates/ovl_select/z_select.c

```diff
@@ -1,40 +1,933 @@
+            day = "\x8Dｻｲｼｮﾉﾋ";
+            break;
+    }
+
+    GfxPrint_Printf(printer, "Day:\x8D%s", day);
+}
+
+void Select_DrawMenu(SelectContext* this) {
+    GraphicsContext* gfxCtx = this->state.gfxCtx;
+    GfxPrint* printer;
+
+    OPEN_DISPS(gfxCtx);
+
+    func_8012C4C0(gfxCtx);
+
+    printer = alloca(sizeof(GfxPrint));
+    GfxPrint_Init(printer);
+    GfxPrint_Open(printer, POLY_OPA_DISP);
+
+    Select_PrintMenu(this, printer);
+    Select_PrintAgeSetting(this, printer, ((void)0, gSaveContext.playerForm));
+    Select_PrintCutsceneSetting(this, printer, ((void)0, gSaveContext.cutscene));
+
+    POLY_OPA_DISP = GfxPrint_Close(printer);
+    GfxPrint_Destroy(printer);
+
+    CLOSE_DISPS(gfxCtx);
+}
+
+void Select_DrawLoadingScreen(SelectContext* this) {
+    GraphicsContext* gfxCtx = this->state.gfxCtx;
+    GfxPrint printer;
+
+    OPEN_DISPS(gfxCtx);
+
+    func_8012C4C0(gfxCtx);
+
+    GfxPrint_Init(&printer);
+    GfxPrint_Open(&printer, POLY_OPA_DISP);
+
+    Select_PrintLoadingMessage(this, &printer);
+
+    POLY_OPA_DISP = GfxPrint_Close(&printer);
+    GfxPrint_Destroy(&printer);
+
+    CLOSE_DISPS(gfxCtx);
+}
+
+void Select_Draw(SelectContext* this) {
+    GraphicsContext* gfxCtx = this->state.gfxCtx;
+
+    func_8012CF0C(gfxCtx, 1, 1, 0, 0, 0);
+
+    SET_FULLSCREEN_VIEWPORT(&this->view);
+    View_RenderView(&this->view, 0xF);
+    if (!this->state.running) {
+        Select_DrawLoadingScreen(this);
+    } else {
+        Select_DrawMenu(this);
+    }
+}
+
+void Select_Main(GameState* thisx) {
+    SelectContext* this = (SelectContext*)thisx;
+
+    Select_UpdateMenu(this);
+    Select_Draw(this);
+}
+
+void Select_Destroy(GameState* thisx) {
+    ShrinkWindow_Fini();
+}
+
+void Select_Init(GameState* thisx) {
+    SelectContext* this = (SelectContext*)thisx;
+
+    this->state.main = Select_Main;
+    this->state.destroy = Select_Destroy;
+
+    this->scenes = sScenes;
+    this->topDisplayedScene = 0;
+    this->currentScene = 0;
+    this->pageDownStops[0] = 0;
+    this->pageDownStops[1] = 19;
+    this->pageDownStops[2] = 37;
+    this->pageDownStops[3] = 51;
+    this->pageDownStops[4] = 59;
+    this->pageDownStops[5] = 73;
+    this->pageDownStops[6] = 91;
+    this->pageDownIndex = 0;
+    this->opt = 0;
+    this->count = ARRAY_COUNT(sScenes);
+
+    ShrinkWindow_Init();
+    View_Init(&this->view, this->state.gfxCtx);
+    this->view.flags = (0x08 | 0x02);
+    this->verticalInputAccumulator = 0;
+    this->verticalInput = 0;
+    this->timerUp = 0;
+    this->timerDown = 0;
+    this->lockUp = false;
+    this->lockDown = false;
+    this->unk_274 = 0;
+
+    if ((dREG(80) >= 0) && (dREG(80) < this->count)) {
+        this->currentScene = dREG(80);
+        this->topDisplayedScene = dREG(81);
+        this->pageDownIndex = dREG(82);
+    }
+
+    Game_SetFramerateDivisor(&this->state, 1);
+    gSaveContext.cutscene = 0;
+    gSaveContext.playerForm = PLAYER_FORM_HUMAN;
+    gSaveContext.linkAge = 0;
```

Comment:
I've heard of 5 and 17, and MM has some new ones, but 0...

Reply:
I think that's how `linkAge` works. 
As a reference from OoT: <https://github.com/zeldaret/oot/blob/b9ad6aac524242a633927f8e0bfc426c6d4f87cb/src/overlays/gamestates/ovl_select/z_select.c#L192>

Reply:
Wonder if it's used in any other ways in MM.